### PR TITLE
Feature/10349 crypto functions part1/10435 rsa oaep with sha256

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
Added RSA OAEP with SHA256 to ENASecurity incl. recommended tests

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10435

### CODE REVIEW

File: src/xcode/ENA/ENASecurity/Sources/ENASecurity/RSAEncryption.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+              }
+        return .success(decodedData)
+    }
+
+    // MARK: - Private
+
+    private let publicKey: Data
+    private let privateKey: Data
+
+    private var publicSecKey: SecKey? {
```
Comment (marcussc) [inline src/xcode/ENA/ENASecurity/Sources/ENASecurity/RSAEncryption.swift:89]: Why not init this struct with the `SecKey` directly? Would remove the optionals. 
An other Idea would be to have an failable init with `Data` that fails if `SecKey` can't be generated from `Data`

File: src/xcode/ENA/ENASecurity/Sources/ENASecurity/RSAEncryption.swift [hunk 2]
```diff
@@ -0,0 +1,108 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import CryptoSwift
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENASecurity/Sources/ENASecurity/RSAEncryption.swift:6]: Is this import needed?

File: src/xcode/ENA/ENASecurity/Sources/ENASecurity/RSAEncryption.swift [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    public func decrypt(data: Data) -> Result<Data, RSAEncryptionError> {
+        // check if keys are available
+        guard let publicKey = publicSecKey else {
+            return .failure(.unknown("public key is missing"))
+        }
+        guard let privateKey = privateSecKey else {
+            return .failure(.unknown("private key is missing"))
+        }
+        // check algorithm is supported
+        guard SecKeyIsAlgorithmSupported(publicKey, .encrypt, SecKeyAlgorithm.rsaEncryptionOAEPSHA256) else {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENASecurity/Sources/ENASecurity/RSAEncryption.swift:64]:
```suggestion
        guard SecKeyIsAlgorithmSupported(publicKey, .decrypt, SecKeyAlgorithm.rsaEncryptionOAEPSHA256) else {
```

File: src/xcode/ENA/ENASecurity/Sources/ENASecurity/RSAEncryption.swift [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            return .failure(.RSA_ENC_NOT_SUPPORTED)
+        }
+        // let's try to decrypt cipher
+        var error: Unmanaged<CFError>?
+        guard let decodedData = SecKeyCreateDecryptedData(
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENASecurity/Sources/ENASecurity/RSAEncryption.swift:69]: The guard is a little bit hard to read. I would prefer to have the call of `SecKeyCreateDecryptedData` separated, if possible.

**Review by ArturFriesen — CHANGES_REQUESTED** [review]

**Review by ArturFriesen — APPROVED** [review]

**Review by marcussc — APPROVED** [review]

**Pull request merged**